lib/features/give/pages/bt_scan_page.dart

```diff
@@ -0,0 +1,185 @@
+      /// and a service UUID.
+      if (scanResult.device.name.isEmpty) {
+        continue;
+      }
+      if (scanResult.advertisementData.manufacturerData.isNotEmpty) {
+        continue;
+      }
+      if (scanResult.advertisementData.serviceData.isEmpty) {
+        continue;
+      }
+      if (scanResult.advertisementData.serviceUuids.isEmpty) {
+        continue;
+      }
+      if (!scanResult.advertisementData.serviceData
+          .containsKey(scanResult.advertisementData.serviceUuids.first)) {
+        continue;
+      }
+
+      if (scanResult.rssi < -75 || scanResult.rssi < -67) {
+        continue;
+      }
+
+      if (!mounted) {
+        return;
+      }
+      context.read<GiveBloc>().add(
+            GiveBTBeaconScanned(
+              userGUID:
+                  (context.read<AuthCubit>().state as AuthSuccess).user.guid,
+              macAddress: scanResult.device.id.toString(),
+              rssi: scanResult.rssi,
+              serviceUUID: scanResult.advertisementData.serviceUuids.first,
+              serviceData: scanResult.advertisementData.serviceData,
+            ),
+          );
+    }
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    flutterBlue.stopScan();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final locals = context.l10n;
+    return Scaffold(
+      appBar: AppBar(
+        leading: const BackButton(),
+        title: Text(locals.giveWithYourPhone),
+      ),
+      body: Center(
+        child: BlocConsumer<GiveBloc, GiveState>(
+          listener: (context, state) {
+            if (state.status == GiveStatus.readyToGive) {
+              Navigator.of(context).pushReplacement(
+                MaterialPageRoute<void>(
+                  builder: (_) => BlocProvider.value(
+                    value: context.read<GiveBloc>(),
+                    child: const GivingPage(),
+                  ),
+                  fullscreenDialog: true,
+                ),
+              );
+            }
+          },
+          builder: (context, state) {
+            return Column(
+              children: [
+                const SizedBox(height: 50),
+                Text(
+                  locals.makeContact,
+                  style: Theme.of(context).textTheme.titleMedium,
+                  textAlign: TextAlign.center,
+                ),
+                const SizedBox(height: 50),
+                Container(
+                  padding: const EdgeInsets.all(10),
+                  child: Image.asset('assets/images/givt_animation.gif'),
+                ),
+                Expanded(child: Container()),
+                Visibility(
+                  visible: _isVisible,
+                  child: Padding(
+                    padding: const EdgeInsets.all(20),
+                    child: ElevatedButton(
+                      onPressed: () => Navigator.of(context).push(
+                        MaterialPageRoute<void>(
```

Comment:
maybe this is bc of lack of my bloc experience, but I do not understand the code in the push method

Reply:
If the bloc is not available in other screens, where different BuildContext is created. Unless the BLoC is added at the MaterialApp widgets, as you have it in the Kids app. So to make it available in other screen i have to provide it to make it accessible. When i pop the screen the bloc will be automatically disposed instead of having a reset method and have it globally available.